src/main/java/rtg/event/EventManagerRTG.java

```diff
@@ -107,22 +104,27 @@ public void generateMinableRTG(OreGenEvent.GenerateMinable event) {
                     return;
 
                 default:
-                	return;
+                	break;
             }
         }
     }
 
     public class InitBiomeGensRTG
     {
-        InitBiomeGensRTG()
-        {
-            Logger.debug(EVENT_SYSTEM + "Initialising InitBiomeGensRTG");
+        InitBiomeGensRTG() {
+            logEventMessage("Initialising InitBiomeGensRTG...");
         }
 
         @SubscribeEvent
         public void initBiomeGensRTG(WorldTypeEvent.InitBiomeGens event) {
 
-            if (event.newBiomeGens[0].getClass().getName().contains("GenLayerEB")) return;
+            if (!(event.worldType instanceof WorldTypeRTG)) {
+                return;
```

Comment:
@whichonespink44 
Because you are now registering all of the event handlers during pre-init, they will all be registered at this point if you are creating a world that is not a RTG WorldType.

`return`ing here only prevents the river remover from being initialised, but we also need to unregister our other event handlers so that we don't interfere in a non-RTG world.

```java
    if (!(event.worldType instanceof WorldTypeRTG)) {
        unRegisterEventHandlers();
        return;
    }
```